google-cloud-spanner/lib/google/cloud/spanner/client.rb

```diff
@@ -371,19 +375,35 @@ def query_options
         #     puts "User #{row[:id]} is #{row[:name]}"
         #   end
         #
+        # @example Query using tag for request query statistics collection.
+        #   require "google/cloud/spanner"
+        #
+        #   spanner = Google::Cloud::Spanner.new
+        #
+        #   db = spanner.client "my-instance", "my-database"
+        #
+        #   results = db.execute_query "SELECT * FROM users", tag: "Tag-1"
+        #
+        #   results.rows.each do |row|
+        #     puts "User #{row[:id]} is #{row[:name]}"
+        #   end
+        #
         def execute_query sql, params: nil, types: nil, single_use: nil,
-                          query_options: nil, call_options: nil
+                          query_options: nil, tag: nil,
+                          call_options: nil
           validate_single_use_args! single_use
           ensure_service!
 
           params, types = Convert.to_input_params_and_types params, types
-
           single_use_tx = single_use_transaction single_use
+          request_options = build_request_options request_tag: tag
```

Comment:
What if this `execute_query` is part of a transaction? How does the transaction tag get passed in?

Reply:
Client can execute only single use transaction, so here only request tag will work. Transaction tag on `execute_query` is  works using transaction object. 

```ruby
db.transaction tag: "Tag-1" do |tx|
  results = tx.execute_query "SELECT * FROM users", tag: "Tag-1-1"
end
```

Implementation ref: https://github.com/googleapis/google-cloud-ruby/blob/880f64fe6db2d48ef989fad356a21d23550cbd88/google-cloud-spanner/lib/google/cloud/spanner/client.rb#L1352-L1373

Here on line 1364 `truncation_tag` will set and pass to transaction object to add queries executed by transaction.

Reply:
Sorry, just to verify I understood it correctly. On the example you provided, the transaction tag will be `Tag-1` and the request tag will be `Tag-1-1`, right?

Reply:
Yes "tag-1" is transaction tag and "Tag-1-1" is request tag.

Reply:
Thanks for clarifying.